File: server/View/Building.cs
using System;
using Newtonsoft.Json;
using Com.GitHub.GlowingPotato.IslandDecimation.Server.Model;

namespace Com.GitHub.GlowingPotato.IslandDecimation.Server.View {
    public class Building : IViewModel {
        public bool HasUpdated {
            get {
                return TypeHasUpdated || Position.HasUpdated || LevelHasUpdated;
            }
        }

#region Type
        BuildingType type;
        bool TypeHasUpdated = true;

        [JsonProperty("type")]
        public BuildingType Type {
            get {
                return type;
            }
            set {
                type = value;
                TypeHasUpdated = true;
            }
        }

        public bool ShouldSerializeType() {
            bool val = TypeHasUpdated;
            if (val) {
                TypeHasUpdated = false;
            }
            return val;
        }
#endregion
#region Position
        [JsonProperty("pos")]
        public Vertex2 Position;

        public bool ShouldSerializePosition() {
            return Position.HasUpdated;
        }
#endregion
#region Level
        int level;
        bool LevelHasUpdated = true;

        [JsonProperty("level")]
        public int Level {
            get {
                return level;
            }
            set {
                level = value;
                LevelHasUpdated = true;
            }
        }

        public bool ShouldSerializeLevel() {
            bool val = LevelHasUpdated;
            if (val) {
                LevelHasUpdated = false;
            }
            return val;
        }
#endregion
    }
}

File: client/src/main/java/com/github/glowingpotato/islanddecimation/view/Building.java
package com.github.glowingpotato.islanddecimation.view;

import com.glutilities.util.Vertex2;

public class Building {
	
	private BuildingType buildingType;
	private Vertex2 position;
	private Integer level;
	
	public Building(BuildingType buildingType, Vertex2 position, Integer level) {
		this.buildingType = buildingType;
		this.position = position;
		this.level = level;
	}
	
	public void update(Building building) {
		if(building.buildingType != null)
			setBuildingType(building.getBuildingType());
		if(building.getPosition() != null)
			setPosition(building.getPosition());
		if(building.getLevel() != null)
			setLevel(building.getLevel());
	}
	
	public BuildingType getBuildingType() {
		return buildingType;
	}

	public void setBuildingType(BuildingType buildingType) {
		this.buildingType = buildingType;
	}

	public Vertex2 getPosition() {
		return position;
	}

	public void setPosition(Vertex2 position) {
		this.position = position;
	}

	public Integer getLevel() {
		return level;
	}

	public void setLevel(Integer level) {
		this.level = level;
	}
}
